Library/Homebrew/extend/ENV/super.rb

```diff
@@ -68,6 +68,10 @@ def setup_build_environment(formula = nil)
     self["HOMEBREW_INCLUDE_PATHS"] = determine_include_paths
     self["HOMEBREW_LIBRARY_PATHS"] = determine_library_paths
     self["HOMEBREW_DEPENDENCIES"] = determine_dependencies
+    unless formula.nil?
+      self["HOMEBREW_FORMULA"] = formula.name
+      self["HOMEBREW_FORMULA_VERSION"] = formula.version
```

Comment:
@xu-cheng: You're right, it should be `pkg_version` or just the formula prefix. Probably the prefix, since that's really what we care about, and that avoids duplicating path construction logic. Thanks for catching; I'll put in another PR for that.